### ISSUE TITLE
Support "trusted profile assume" with a go authenticator.

### ISSUE THREAD

**Issue by sjhx** [IBM/go-sdk-core#226]
**Is your feature request related to a problem? Please describe.**
missing feature I believe at least in the go sdk

**Describe the solution you'd like**
Essentially do this:
```shell
curl -X POST "https://iam.cloud.ibm.com/identity/token"\
  --header "Content-Type: application/x-www-form-urlencoded"\
  --data-urlencode 'grant_type=urn:ibm:params:oauth:grant-type:assume'\
  --data-urlencode 'access_token=ACCESS-TOKEN'\
  --data-urlencode 'profile_id=ProfileId-123123123'
```

**Describe alternatives you've considered**
unable to see how to do this with the SDK publisjed

**Additional context**
I could not find a way with the provided SDK to assume a trusted profile.
I ended up writing an authenticator to wrap and IAMAuthenticator (containing service apikey) which generates a token for the trusted profile that the apikey is allowed to assume. I think this should be provided.

**Comment by padamstx** [comment]
Looking into this... will consult with IAM team for more info.

**Comment by padamstx** [comment]
@sjhx Stuart, could you please describe your particular use-case in a bit more detail?  I assume you have a Go application that uses a Go SDK and you need to use this new IAM feature to "assume" the identity of the trusted profile (sort of like sudo on linux).

**Comment by dpopp07** [comment]
This feature was released in [v5.18.0](https://github.com/IBM/go-sdk-core/releases/tag/v5.18.0).
